### PR TITLE
fix NLP zero division error

### PR DESCRIPTION
**Pull request by liususan091219**
## Why are these changes needed?

Fix the NLP zero division error in https://github.com/microsoft/FLAML/actions/runs/4783351775/jobs/8503550377#step:13:145

## Checks

- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by sonichi — COMMENTED** [review]
Is this PR ready for review? The title suggests not done but the PR is not a draft.

File: flaml/automl/model.py [hunk 1]
```diff
@@ -1070,6 +1070,7 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
                 task=self._task,
                 y_pred=predictions,
                 tokenizer=self.tokenizer,
+                X_len=len(y_true),
```
Comment (sonichi) [inline flaml/automl/model.py:1073]: remove

File: flaml/automl/model.py [hunk 2]
```diff
@@ -1171,11 +1172,12 @@ def predict(self, X, **pred_kwargs):
             predictions = new_trainer.predict(test_dataset, **kwargs).predictions
         except ZeroDivisionError:
             logger.warning("Zero division error appeared in HuggingFace Transformers.")
-            predictions = np.array([[-0.05, -0.05]] * len(test_dataset))
+            predictions = None  # np.array([[-0.05, -0.05]] * len(test_dataset))
         post_y_pred, _ = postprocess_prediction_and_true(
             task=self._task,
             y_pred=predictions,
             tokenizer=self.tokenizer,
+            X_len=len(X),
```
Comment (sonichi) [inline flaml/automl/model.py:1180]: remove

File: flaml/automl/nlp/huggingface/utils.py [hunk 3]
```diff
@@ -309,8 +309,10 @@ def tokenize_swag(this_row, tokenizer, hf_args=None, return_column_name=False):
         return [tokenized_example[x] for x in tmp_column_names]
 
 
-def postprocess_prediction_and_true(task, y_pred, tokenizer, hf_args, y_true=None, X=None):
+def postprocess_prediction_and_true(task, y_pred, tokenizer, X_len, hf_args, y_true=None, X=None):
     # postprocess the matrix prediction y_pred and ground truth y_true into user readable format, e.g., for summarization, decode into text
+    if y_pred is None:
+        return np.array([0] * X_len)
```
Comment (sonichi) [inline flaml/automl/nlp/huggingface/utils.py]:
```suggestion
        return np.array([0] * len(X))
```

**Review by sonichi — COMMENTED** [review]
Suggested a simpler implementation.

**Review by sonichi — APPROVED** [review]

**Review by thinkall — APPROVED** [review]

**Pull request merged**